coreax/weights.py

```diff
@@ -52,9 +52,6 @@ class WeightsOptimiser(ABC):
 
     def __init__(self, kernel: coreax.kernel.Kernel) -> None:
         """Initialise a weights optimiser class."""
-        coreax.validation.validate_is_instance(
-            x=kernel, object_name="kernel", expected_type=coreax.kernel.Kernel
```

Comment:
Just to check, the new validation is meant to remove all these kinds of instance checks (where the type hint is telling you the correct type to pass)?

Reply:
Yes - we should not see the validation functions being used anywhere. We might only see an instance check if we want to raise a helpful error message to guide the user on what to fix

Reply:
(And this should be using isinstance(object, type) rather than our own functions